deeplake/util/connect_dataset.py

```diff
@@ -0,0 +1,68 @@
+from typing import Tuple, Optional
+from deeplake.client.client import DeepLakeBackendClient
+from deeplake.client.log import logger
+from deeplake.util.path import is_hub_cloud_path, get_path_type, get_org_id_and_ds_name
+from deeplake.util.exceptions import InvalidDestinationPathError, InvalidSourcePathError
+from deeplake.util.logging import log_visualizer_link
+
+
+def log_dataset_connection_success(ds_path: str):
+    logger.info("Dataset connected successfully.")
+    log_visualizer_link(ds_path)
+
+
+def connect_dataset_entry(
+    src_path: str,
+    creds_key: str,
+    dest_path: Optional[str],
+    org_id: Optional[str],
+    ds_name: Optional[str],
+    token: Optional[str],
+) -> str:
+    if is_hub_cloud_path(src_path):
+        raise InvalidSourcePathError(
+            "Source dataset is already accessible by a Deep Lake path."
+        )
+
+    if not is_path_connectable(src_path):
+        raise InvalidSourcePathError(
+            f"Source path may only be an s3 or gcs path. Got {src_path}."
+        )
+
+    client = DeepLakeBackendClient(token)
+    org_id, ds_name = _get_org_id_and_ds_name(
+        dest_path=dest_path, org_id=org_id, ds_name=ds_name
+    )
+
+    connected_id = client.connect_dataset_entry(
+        src_path=src_path, org_id=org_id, ds_name=ds_name, creds_key=creds_key
+    )
+
+    return connected_id
+
+
+def is_path_connectable(path: str) -> bool:
+    return get_path_type(path) in ("s3", "gcs")
+
+
```

Comment:
The same applies here. We can create a class that will wrap all information that is needed, and later retrieve them in the method.

```suggestion

class DsInfo:
    def __init__(self, dest_path, org_id, ds_name):
        self.dest_path = dest_path
        self.org_id = org_id
        self.ds_name = ds_name
        
    def check_errors(self):
        if self.org_id is None:
            if self.dest_path is None:
                raise InvalidDestinationPathError(
                    "Invalid destination path. Either the organization or the destination path must be provided."
                )

             if not is_hub_cloud_path(self.dest_path):
                 raise InvalidDestinationPathError(
                     "Destination path must be a Deep Lake path."
                )
    
    def get_org_id_and_ds_name(self):
        if self.org_id is None:
            self.org_id, self.ds_name = get_org_id_and_ds_name(self.dest_path)
        return self.org_id, self.ds_name
```

Reply:
Yep, this way it may be reused potentially :D